### PR TITLE
Check fake user

### PR DESCRIPTION
**Pull request by vicdoval**
Fixes a long standing issue of being able to lose your work if you forget to click in the fake user toggle.

- [x] Ready for merge.

### CODE REVIEW

File: core/update_system.py [hunk 1]
```diff
@@ -476,6 +479,7 @@ def process_from_nodes(nodes):
             print("Something not very important happend in Blender memory", node, type(node))
 
     ng = nodes[0].id_data
+    check_fake_user(ng)
```
Comment (Durman) [inline core/update_system.py:482]:
```suggestion
    ng.use_fake_user = True
```

**Comment by Durman** [comment]
And there is no much sense to have this buttons then I guess.

File: core/update_system.py (hunk 1 above)
Comment (vicdoval) [reply core/update_system.py:482]: Are you sure that would be something to do on every update? thats why a i added the check, I dont know if something else happens inside blender when use_fake_user is set

Comment (Durman) [reply core/update_system.py:482]: I think it is static property which is taking into account when Blender makes decisions of clearing data.

Comment (Durman) [reply core/update_system.py:482]: Probably even ban could be done. We can override creating new tree operator.

Or we can try to subscribe to `node_tree` attribute changes of a windows manager.

Comment (Durman) [reply core/update_system.py:482]: Yes subscribesion works
```Python
bpy.msgbus.subscribe_rna(
        key=(bpy.types.SpaceNodeEditor, 'node_tree'),
        owner=object(),
        args=(),
        notify=lambda *args: print(f'Tree was changed, probably new one was created')
    )
```

Comment (vicdoval) [reply core/update_system.py:482]: @Durman Should I close this PR and let you do it in a more efficient way?

**Pull request closed without merging**